lib/projects/__tests__/structure.test.ts

```diff
@@ -0,0 +1,274 @@
+import fs from 'fs';
+import * as HSfs from '@hubspot/local-dev-lib/fs';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import {
+  getComponentTypeFromConfigFile,
+  loadConfigFile,
+  getAppCardConfigs,
+  getIsLegacyApp,
+  componentIsApp,
+  findProjectComponents,
+  getProjectComponentTypes,
+  getComponentUid,
+  componentIsPublicApp,
+} from '../structure';
+import { ComponentTypes, Component } from '../../../types/Projects';
+
+jest.mock('fs');
+jest.mock('@hubspot/local-dev-lib/fs');
+jest.mock('@hubspot/local-dev-lib/logger');
+
+const mockedReadFileSync = fs.readFileSync as jest.Mock;
+const mockedWalk = HSfs.walk as jest.Mock;
+
+const getMockPrivateAppConfig = (cards: Array<{ file: string }> = []) => ({
+  name: 'test-app',
+  description: 'test-description',
+  uid: 'test-uid',
+  scopes: ['test-scope'],
+  public: true,
+  extensions: {
+    crm: {
+      cards,
+    },
+  },
+});
+
+describe('lib/projects/structure', () => {
+  describe('getComponentTypeFromConfigFile()', () => {
+    it('returns correct type for public app config', () => {
+      expect(getComponentTypeFromConfigFile('public-app.json')).toBe(
+        ComponentTypes.PublicApp
+      );
+    });
+
+    it('returns correct type for private app config', () => {
+      expect(getComponentTypeFromConfigFile('app.json')).toBe(
+        ComponentTypes.PrivateApp
+      );
+    });
+
+    it('returns correct type for theme config', () => {
+      expect(getComponentTypeFromConfigFile('theme.json')).toBe(
+        ComponentTypes.HublTheme
+      );
+    });
+
+    it('returns null for unknown config file', () => {
+      expect(getComponentTypeFromConfigFile('unknown.json')).toBeNull();
+    });
+  });
+
+  describe('loadConfigFile()', () => {
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
```

Comment:
Nit: This should be globally set to reset after every test in the jest config file.

Reply:
Cool, I didn't know we had that. I can probably remove other instances of this that I've added in other PR's